acrg/name/name.py

```diff
@@ -1562,30 +1563,34 @@ def local_solar_time(dataset):
         hours = dataset.time.to_pandas().index.hour
         return hours
     
-    def local_ratio(dataset):
+    @numba.jit()
+    def assign_localness(dataset,wh_rlat,wh_rlon):
+        local_sum=np.zeros(dataset.shape[2])
```

Comment:
I'm surprised this expression `dataset.shape[2]` works - if this is a Dataset I'm not sure this has a generic `.shape` attribute and if this will always be a DataArray then the naming is a little confusing.

(A more generic alternative for a Dataset could be to use the `.dims` attribute (e.g. something like `dataset.dims[dim]`))

Reply:
From looking at the code below looks like the input for `dataset` is actually a numpy array:
```
fp_data_H_site_fp = fp_data_H[site].fp.values
local_sum = assign_localness(fp_data_H_site_fp,wh_rlat,wh_rlon)
```

So I think just renaming `dataset` to something like `array` may make this more clear what will work and what won't.